### DISCUSSION TITLE
str repr dunders

### DISCUSSION THREAD

**Discussion by PabloRuizCuevas** [staircase-dev/staircase#172] [Ideas]
Hi! just saw that you include me in the change-log of the new versions, thanks! :) I will probably continue using this library on the time being, and i was wondering if we could change a bit these methods to give a bit more of information of the object itselfs

    def __str__(self) -> str:
        """
        Return str(self)
        """
        return f"<staircase.{self.class_name}, id={id(self)}>"

    def __repr__(self) -> str:
        """
        Return string representation of Stairs
        """
        return str(self)
        
 I'm not yet super familiar with all the internals of the library but for me I would find more useful to make __repr__ to return self.to_frame()
 or self._data
 
 In general i would like that when passing a stairs object in a jupyter notebook it prints something useful by default with no need of calling to_frame() or print, but ofc this is a personal preference.I imagine you may have reasons to let it as is, just writing ideas here in case you think they are good ones.

**Comment by venaturum** [comment]
Hey Pablo,

Sorry for not getting to this earlier, a bit flat out at the moment.  I probably need to think about it in detail a little more before we make a permanent change, but you can monkeypatch a solution for yourself in the meantime:

```
import staircase as sc

def my_repr(self):
    return str(self.to_frame())

sc.Stairs.__repr__ = my_repr
```

Cheers, Riley.

**Reply by PabloRuizCuevas** [reply]
hi, no worries, take your time. and sure, this is  just an idea, I may be putting some more ideas in the following days, feel free to disregard them if you don't think they are the best, or if they will put major backwards compatibility issues.
